src/LibUvLoop.php

```diff
@@ -0,0 +1,293 @@
+    private $signals;
+    private $signalEvents = array();
+    private $streamListener;
+
+    public function __construct()
+    {
+        if (!function_exists('uv_loop_new')) {
+            throw new BadMethodCallException('Cannot create LibUvLoop, ext-uv extension missing');
+        }
+
+        $this->uv = \uv_loop_new();
+        $this->futureTickQueue = new FutureTickQueue();
+        $this->timerEvents = new SplObjectStorage();
+        $this->streamListener = $this->createStreamListener();
+        $this->signals = new SignalsHandler();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addReadStream($stream, $listener)
+    {
+        if (isset($this->listeners[(int) $stream]['read'])) {
+            return;
+        }
+
+        $this->listeners[(int) $stream]['read'] = $listener;
+        $this->addStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addWriteStream($stream, $listener)
+    {
+        if (isset($this->listeners[(int) $stream]['write'])) {
+            return;
+        }
+
+        $this->listeners[(int) $stream]['write'] = $listener;
+        $this->addStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeReadStream($stream)
+    {
+        if (!isset($this->streamEvents[(int) $stream])) {
+            return;
+        }
+
+        unset($this->listeners[(int) $stream]['read']);
+        $this->removeStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function removeWriteStream($stream)
+    {
+        if (!isset($this->streamEvents[(int) $stream])) {
+            return;
+        }
+
+        unset($this->listeners[(int) $stream]['write']);
+        $this->removeStream($stream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addTimer($interval, $callback)
+    {
+        $timer = new Timer( $interval, $callback, false);
+
+        $that = $this;
+        $timers = $this->timerEvents;
+        $callback = function () use ($timer, $timers, $that) {
+            call_user_func($timer->getCallback(), $timer);
+
+            if ($timers->contains($timer)) {
+                $that->cancelTimer($timer);
+            }
+        };
+
+        $event = \uv_timer_init($this->uv);
+        $this->timerEvents->attach($timer, $event);
+        \uv_timer_start(
+            $event,
+            (int)($interval * 1000),
+            0,
+            $callback
+        );
+
+        return $timer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addPeriodicTimer($interval, $callback)
+    {
+        $timer = new Timer($interval, $callback, true);
+
+        $callback = function () use ($timer) {
+            call_user_func($timer->getCallback(), $timer);
+        };
+
+        $event = \uv_timer_init($this->uv);
+        $this->timerEvents->attach($timer, $event);
+        \uv_timer_start(
+            $event,
+            $interval * 1000,
+            $interval * 1000,
```

Comment:
The cast to integer is required because on float it errors with: `uv_timer_start() expects parameter 2 to be integer, float given`

Also added `ceil()`, the difference with a only a cast is that the cast behaves like `floor()`. So with `ceil()` we enforce the timer runs after it's intended run time where `floor()` might run it just before. But this is at the milisecond level so it won't matter that much